resources/prosody-plugins/mod_av_moderation_component.lua

```diff
@@ -0,0 +1,198 @@
+local get_room_by_name_and_subdomain = module:require 'util'.get_room_by_name_and_subdomain;
+local is_healthcheck_room = module:require 'util'.is_healthcheck_room;
+local json = require 'util.json';
+local st = require 'util.stanza';
+local um_is_admin = require "core.usermanager".is_admin;
+
+local muc_component_host = module:get_option_string('muc_component');
+if muc_component_host == nil then
+    log('error', 'No muc_component specified. No muc to operate on!');
+    return;
+end
+
+module:log('info', 'Starting av_moderation for %s', muc_component_host);
+
+local function is_admin(jid)
+    return um_is_admin(jid, module.host);
+end
+
+-- Notifies that av moderation has been enabled or disabled
+-- @param jid the jid to notify, if missing will notify all occupants
+function notify_occupants_enable(jid, enable, room)
+    local body_json = {};
+    body_json.type = 'av_moderation';
+    body_json.enabled = enable;
+    body_json.room = room.jid;
+    local body_json_str = json.encode(body_json);
+
+    local notify = function(jid_to_notify)
+        local stanza = st.message({ from = module.host; to = jid_to_notify; })
+                :tag('json-message', { xmlns = 'http://jitsi.org/jitmeet' }):text(body_json_str):up();
+        module:send(stanza);
+    end
+
+    if jid then
+        notify(jid);
+    else
+        for _, occupant in room:each_occupant() do
+            notify(occupant.jid);
+        end
+    end
+
+end
+
+-- Notifies about a jid added to the whitelist. Notifies all moderators and admin and the jid itself
+-- @param jid the jid to notify about the change
+-- @param moderators whether to notify all moderators in the room
+-- @param room the room where to send it
+function notify_whitelist_change(jid, moderators, room)
+    local body_json = {};
+    body_json.type = 'av_moderation';
+    body_json.whitelists = room.av_moderation;
+    body_json.room = room.jid;
+    local body_json_str = json.encode(body_json);
+
+    for _, occupant in room:each_occupant() do
+        if (moderators and occupant.role == 'moderator') or occupant.jid == jid then
+            local stanza = st.message({
+                from = module.host; to = occupant.jid; }):tag('json-message',{ xmlns = 'http://jitsi.org/jitmeet' })
+                    :text(body_json_str):up();
+            module:send(stanza);
+        end
+    end
+
+end
+
+-- receives messages from client currently connected to the room
+-- clients indicates their own dominant speaker events
+function on_message(event)
+    local session = event.origin;
+
+    -- Check the type of the incoming stanza to avoid loops:
+    if event.stanza.attr.type == 'error' then
```

Comment:
Should you also include `result` here?

Reply:
Nope, this just skips processing here, but continue executing on the other listeners for this event. In the other case it results an error and stops the processing of this message.
`Returning any value other than nil will halt processing of the event, and return that value to the code that fired the event.`

Reply:
I don't get it. Why do you want to continue handling stanzas of type `result`? They shouldn't be responded to either (just like `error`), right?